drivers/hmis/app/graphql/types/hmis_schema/client.rb

```diff
@@ -93,25 +93,30 @@ def self.configuration
       transform_changes: ->(version, changes) do
         result = changes
         [
-          ['race', Hmis::Hud::Client.race_enum_map, :RaceNone],
-          ['gender', Hmis::Hud::Client.gender_enum_map, :GenderNone],
+          ['race', Hmis::Hud::Client.race_enum_map, 'RaceNone'],
+          ['gender', Hmis::Hud::Client.gender_enum_map, 'GenderNone'],
         ].each do |input_field, enum_map, none_field|
           relevant_fields = [*enum_map.base_members.map { |member| member[:key].to_s }, none_field.to_s, input_field]
           next unless changes.slice(*relevant_fields).present?
 
           result = result.except(*relevant_fields)
-          old_client = version.reify
 
-          # Reify the next version to get next values; If no next version, then we're at the latest update and the current object will have the next values
-          new_client =  version.next&.reify || version.item
-
-          old_value = { input_field => nil }
-          new_value = { input_field => nil }
-
-          old_value = Hmis::Hud::Processors::ClientProcessor.multi_fields_to_input(old_client, input_field, enum_map, none_field) if old_client.present?
-          new_value = Hmis::Hud::Processors::ClientProcessor.multi_fields_to_input(new_client, input_field, enum_map, none_field) if new_client.present?
-
-          result = result.merge(input_field => [old_value[input_field], new_value[input_field]])
+          # delta = [[old], [new]]
+          delta = [
+            version.object || {},
+            changes.transform_values(&:last),
+          ].map do |doc|
+            none_value = doc[none_field]
+            if none_value.nil? || none_value.zero?
+              enum_map.base_members.
+                filter { |item| doc[item[:key].to_s] == 1 }.
+                map { |item| item[:value] }
+            else
+              [none_value]
+            end
+          end
```

Comment:
I haven't tracked down the cause yet, but I'm not seeing the expected behavior here. If I set a client up with 2 genders, then remove one of them, this is resolving that gender got changed to `null`, even though it got changed to "culturally specific." I confirmed in the version table that the correct change was recorded, but it doesn't appear to be resolved accurately.

Reply:
`changes` is just the `version.object_changes`, it looks like (see base audit event). Why are we using `version.object` to compare with `version.object_changes.transform_values(&:last)` -- wouldn't they be the same? I think I'm confused..


As a general concern, it would be preferred to continue to show ALL gender (or race) fields in the change history. I can't tell if this code is trying to do that or not.

For example, if someone previously had `Man, Non-Binary` selected, and then removed the `Man` option, the change history should appear as `Gender:     Man, Non-Binary  =>  Non-Binary`. I don't think we can do that without reifying because we need to know the previous value for all genders (since the `Non-Binary` column didn't change, it probably wouldn't be included in object_changes, right?).

If that presents issues, we could display it differently in the interface, like having one "row" per gender. (eg `Man:    True => False`, but I think its less clear.